### PR TITLE
[C-2941] Modify cloudflare worker to pull in SEO data from discovery nodes

### PR DESCRIPTION
**Pull request by raymondjacobson**
### Description

Most (95+%) of pages aren't recognized by google at all right now.
This does a nifty on the fly html rewrite on the cloudflare side (which is then cached) and injects the proper tags from a request sent to discovery nodes.
This is a fairly well supported pattern:
https://developers.cloudflare.com/workers/runtime-apis/html-rewriter/#async-handlers

This will cause a minor slowdown when things aren't cached, but I think well worth it for discovery purposes.

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

Touches main serving.

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

```
npx wrangler preview --watch --env ray
```

Verified page source to see the right tags populated for
* track page
* playlist page / album page
* user page
* home page
* /audio page (just to test some other pages)

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rj-seo

File: packages/web/scripts/workers-site/index.js [hunk 1]
```diff
@@ -17,17 +42,175 @@ addEventListener('fetch', (event) => {
...
+class HeadElementHandler {
+  constructor(pathname) {
+    self.pathname = pathname
+  }
+
+  async element(element) {
+    const { metadata, name } = await getMetadata(self.pathname)
+
+    if (!metadata || !name) {
+      // We did parse this to anything we have custom tags for, so just return the default tags
```
Comment (sliptype) [inline packages/web/scripts/workers-site/index.js:140]: Should this be

```suggestion
      // We didn't parse this to anything we have custom tags for, so just return the default tags
```

**Review by sliptype — APPROVED** [review]
Very cool! Would be good to see how much this slows uncached page loads on average

I mean this is basically SSR for the metadata, I wonder if we could SSR the entire app using cloudflare workers

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rj-seo

**Pull request merged**